plugins/woocommerce/includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -193,18 +193,34 @@ public function update( &$order ) {
 		// Update the order.
 		parent::update( $order );
 
-		// Fire a hook depending on the status - this should be considered a creation if it was previously draft status.
 		$new_status = $order->get_status( 'edit' );
 
-		if ( $new_status !== $previous_status && in_array( $previous_status, array( 'new', 'auto-draft', 'draft', 'checkout-draft' ), true ) ) {
+		// We need to remove the wc- prefix from the status for comparison and proper evaluation of new vs updated orders.
+		if ( strpos( $previous_status, 'wc-' ) === 0 ) {
+			$previous_status = substr( $previous_status, 3 );
+		}
+
+		if ( strpos( $new_status, 'wc-' ) === 0 ) {
```

Comment:
The abstract order class has some code which removes prefix in the `set_status`  method. How about creating a utility method that this can share?

Reply:
Good call, added in https://github.com/woocommerce/woocommerce/commit/8b50abaf6bd7e9f48540559799752e562f75b5db and replaced a few other places in the code that was doing the same thing 👍🏼